multiqc/qualimap/__init__.py

```diff
@@ -0,0 +1,115 @@
+             the quality control of alignment sequencing data and its derivatives like feature counts.</p>'
+        self.analysis_dir = report['analysis_dir']
+        self.output_dir = report['output_dir']
+
+        # Find QualiMap reports
+        qualimap_raw_data = {}
+        for root, dirnames, filenames in os.walk(self.analysis_dir, followlinks=True):
+            if 'genome_results.txt' in filenames and 'raw_data_qualimapReport' in dirnames:
+                with open(os.path.join(root, 'genome_results.txt'), 'r') as gr:
+                    for l in gr:
+                        if 'bam file' in l:
+                            s_name = self.clean_s_name(os.path.basename(l.split(' = ')[-1]))
+
+                if report['prepend_dirs']:
+                    s_name = "{} | {}".format(os.path.dirname(root).replace(os.sep, ' | '), s_name).lstrip('. | ')
+
+                qualimap_raw_data[s_name] = {}
+                qualimap_raw_data[s_name]['reports'] = {os.path.splitext(r)[0]: os.path.join(root, 'raw_data_qualimapReport', r) \
+                    for r in os.listdir(os.path.join(root, 'raw_data_qualimapReport'))}
+                qualimap_raw_data[s_name]['plots'] = {os.path.splitext(r)[0]: os.path.join(root, 'images_qualimapReport', r) \
+                    for r in os.listdir(os.path.join(root, 'images_qualimapReport'))}
+
+        if len(qualimap_raw_data) == 0:
+            logging.debug("Could not find any QualiMap reports in {}".format(self.analysis_dir))
+            raise UserWarning
+
+        logging.info("Found {} QualiMap reports".format(len(qualimap_raw_data)))
+
+        self.sections = list()
+
+        # Section 1 - Coverage Histogram
+        histogram_data = self.qualimap_cov_his(qualimap_raw_data)
+        if len(histogram_data) > 0:
+            self.sections.append({
+                'name': 'Coverage Histogram',
+                'anchor': 'coverage-histogram',
+                'content': self.qualimap_cov_his_plot(histogram_data)
+            })
+
+
+    def qualimap_cov_his(self, qualimap_raw_data):
+        parsed_data = {}
+        for sn, data in qualimap_raw_data.iteritems():
+            cov_report = data['reports'].get('coverage_histogram')
+            if cov_report:
+                counts={}
+                try:
+                    with open(cov_report, 'r') as fh:
+                        next(fh) # skip the header
+                        for line in fh:
+                            (coverage, count) = line.split(None, 1)
+                            coverage = int(round(float(coverage)))
+                            count = float(count)
+                            counts[coverage] = count
+
+                except IOError as e:
+                    logging.error("Could not load input file: {}".format(cov_report))
+                    raise
+                parsed_data[sn] = counts
+
+        return parsed_data
+
+
+    def qualimap_cov_his_plot(self, parsed_data):
```

Comment:
yup, maybe we (you? :-P) should think about this, not only for the plotting, but also all the html code embedded in the code. Is not that I have a better solution right now, but seems like something that could maybe be templated?